python/ml4ir/applications/ranking/model/layers/masking.py

```diff
@@ -98,3 +98,98 @@ def call(self, inputs, mask=None, training=None):
+
+    Example
+    -------
+    x = np.ones((2, 5, 4))
+        array([[[1., 1., 1., 1.],
+                [1., 1., 1., 1.],
+                [1., 1., 1., 1.],
+                [1., 1., 1., 1.],
+                [1., 1., 1., 1.]],
+
+               [[1., 1., 1., 1.],
+                [1., 1., 1., 1.],
+                [1., 1., 1., 1.],
+                [1., 1., 1., 1.],
+                [1., 1., 1., 1.]]])
+    record_feature_mask = RecordFeatureMask(0.5)
+    record_feature_mask(x)
+        array([[[1., 1., 0., 1.],
+                [1., 1., 0., 1.],
+                [1., 1., 0., 1.],
+                [1., 1., 0., 1.],
+                [1., 1., 0., 1.]],
+
+               [[0., 1., 1., 1.],
+                [0., 1., 1., 1.],
+                [0., 1., 1., 1.],
+                [0., 1., 1., 1.],
+                [0., 1., 1., 1.]]])
+    """
+
+    def __init__(self,
+                 name="query_feature_mask",
+                 mask_rate: float = 0.2,
+                 mask_at_inference: bool = False,
+                 requires_mask: bool = False,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        name: str
+            Layer name
+        mask_rate: float
+            Rate at which the records in the query batch need to be masked to 0s
+        mask_at_inference: boolean
+            Whether to also mask at inference
+            Useful for testing performance at inference time, but should be set to False when training a model to deploy
+        requires_mask: bool
+            Indicates if the layer requires a mask to be passed to it during forward pass
+        kwargs:
+            Additional key-value args that will be used for configuring the layer
+        """
+        super().__init__(name=name, **kwargs)
+
+        self.mask_rate = mask_rate
+        self.mask_at_inference = mask_at_inference
+        self.requires_mask = requires_mask
+
+        super(QueryFeatureMask, self).__init__(**kwargs)
+
+    def call(self, inputs, mask=None, training=None):
+        """
+        Invoke the query feature mask layer for the input feature tensor
+
+        Parameters
+        ----------
+        inputs: Tensor object
+            Input ranking feature tensor
+            Shape: [batch_size, sequence_len, feature_dim]
+        mask: Tensor object
+            Mask to be used to identify records to ignore in query (unused in this layer)
+            Shape: [batch_size, sequence_len]
+        training: bool
+            If the layer should be run as training or not
+
+        Returns
+        -------
+        Tensor object
+            Feature tensor where mask_rate of records' features have been masked out (or set to 0.)
+            Shape: [batch_size, sequence_len, feature_dim]
+        """
+        batch_size, sequence_len, feature_dim = inputs.shape
+        if self.mask_at_inference or training:
+            mask = tf.cast(tf.random.uniform([batch_size, feature_dim]) > self.mask_rate, dtype=tf.float32)
```

Comment:
@mohazahran sometimes, tensorflow saves these as constants instead of layers. Worth double checking.

Reply:
@mohazahran can you change the `>` as well with something like https://www.tensorflow.org/api_docs/python/tf/math/greater

Reply:
@mohazahran can you change the > as well with something like https://www.tensorflow.org/api_docs/python/tf/math/greater